docs/how_to/upgrade-tf-controller.md

```diff
@@ -0,0 +1,16 @@
+# Upgrading TF-Controller
+
+Please follow these steps to upgrade TF-Controller:
+
+1. Read the latest release changelogs.
+2. Check your API versions.
+3. Suspend related Kustomization objects to minimize impact on live systems.
+4. Upgrade Flux first, following [the Flux documentation](https://fluxcd.io/flux/installation/upgrade/).
+5. Back up Terraform tfstates to avoid losing data. If you're using the default backend with secrets in Kubernetes, use your backup toolset (i.e., Velero) to back up the state data.
```

Comment:
I would swap these two. If Flux was updated, but the new Flux version is not supported by the currently TF Controller, the controller can behave weirdly and safer to save states before it happens.

For example upgrading from v0.14 to v0.15. That requires a new Flux version. Either way (upgrade flux or tf controller first) it will throw errors if the controller reconciliation loop starts.

4. Suspend terraform resources. (`tfctl suspend --all`)   << this needs to be done before backup to make sure we don't get new state changes.
5. Back up your data.
6. Upgrade Flux
...
N. (probably after current `8.` or `9.`) Resume terraform resources. Either one-by-one for critical resources, or YOLO `tfctl resume --all`

Reply:
What do you think @chanwit ?

Reply:
@yitsushi - Thanks for outlining the additional steps required - I'm going to look at testing these once the list of steps has been confirmed, will keep watching here until they're finalised.

Reply:
Also worth noting here: it might be prudent to include any upgrade steps, specific to a particular release, in the release notes for that release? for example, you've noted:

> For example upgrading from v0.14 to v0.15. That requires a new Flux version

These specifics are best noted in the release notes, IMO - I'm not sure if that's currently the case ... this would answer a question a user had: "How do I know when to update `apiVersion` for TF CRs when upgrading" .. we could potentially point them to specific release notes _per release_ in this instance - I hope that makes sense? :)

Reply:
The example I provided was to support my additions, to show how it can behave weirdly. It shouldn't cause much issues with states, but we definitely want to make backup before upgrading Flux.

Reply:
@chanwit Would you reverse steps 4 and 5 as @yitsushi suggests? LMK, and then I'll wrap this up.

Reply:
@chanwit Yes, I've already been in touch with @madAndroid about testing them

Reply:
@yitsushi @chanwit Updated based on your comments -- thanks, both. @madAndroid, want to give a test run?

Reply:
@chanwit - thanks for the clarification :) is it worth having these details present in the release notes? it would ideally be easily reachable by clients, so that we can refer them to that location when they have questions around apiVersion upgrades

Reply:
Apologies, I do see that the CHANGELOG does include details around breaking changes and the like, so would this be the correct location to point customers? => https://github.com/weaveworks/tf-controller/blob/main/CHANGELOG.md

Reply:
Re: testing, I will make sure the steps are followed ASAP - I have a test cluster I can use for this

Reply:
@madAndroid did you have a chance to test this?